### ISSUE TITLE
-Wmissing-field-initializers error with Python 3.8

### ISSUE THREAD

**Issue by simark** [swig/swig#1701]
Generating some Python bindings, and building against CPython 3.8 with -Wmissing-field-initializers (with either clang or gcc) gives me an error.  For example if I take the example for the tutorial:

```
 /* example.i */
 %module example
 %{
 /* Put header files here or function declarations like below */
 extern double My_variable;
 extern int fact(int n);
 extern int my_mod(int x, int y);
 extern char *get_time();
 %}
 
 extern double My_variable;
 extern int fact(int n);
 extern int my_mod(int x, int y);
 extern char *get_time();
```
and put it in `foo.i`.  I generate the C code with:

    swig -python foo.i

And build with

    gcc foo_wrap.c $(python3.8-config --cflags) -Wmissing-field-initializers -Werror

I get a few errors like:

```
foo_wrap.c: In function ‘SwigPyObject_TypeOnce’:
foo_wrap.c:1742:5: error: missing initializer for field ‘tp_vectorcall’ of ‘PyTypeObject’ {aka ‘const struct _typeobject’} [-Werror=missing-field-initializers]
 1742 |     };
      |     ^
In file included from /usr/include/python3.8/object.h:746,
                 from /usr/include/python3.8/pytime.h:6,
                 from /usr/include/python3.8/Python.h:85,
                 from foo_wrap.c:154:
/usr/include/python3.8/cpython/object.h:257:20: note: ‘tp_vectorcall’ declared here
  257 |     vectorcallfunc tp_vectorcall;
      |                    ^~~~~~~~~~~~~
```

**Comment by vadz** [comment]
AFAICS this should have been fixed by eaa16914d1686002d33cb3f61875832977146805, i.e. SWIG from master should work.

**Comment by simark** [comment]
Huh, my bad for not trying with master first.  This will surely fix it (if not I'll re-open the issue).  Thanks!
